sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -582,34 +598,34 @@ export class KeyClient {
   ): Promise<KeyVaultKey> {
     if (options) {
       const requestOptions = operationOptionsToRequestOptionsBase(options);
-      const { enabled, notBefore, expiresOn: expires, ...remainingOptions } = requestOptions;
-      const unflattenedOptions = {
-        ...remainingOptions,
-        keyAttributes: {
-          enabled,
-          notBefore,
-          expires
-        }
-      };
+      const { keyOps, enabled, notBefore, expiresOn, tags } = requestOptions;
 
-      const span = this.createSpan("updateKeyProperties", unflattenedOptions);
+      const span = this.createSpan("updateKeyProperties", requestOptions);
 
-      let response: UpdateKeyResponse;
+      let response: KeyVaultClientUpdateKeyResponse;
+      const updateKeyParams: UpdateKeyPropertiesOptions = {
+        keyOps,
+        enabled,
+        notBefore,
+        expiresOn,
+        tags
+      };
```

Comment:
I find it interesting how different is this from: https://github.com/Azure/azure-sdk-for-js/pull/9732/files#r446186630